wagtail/images/image_operations.py

```diff
@@ -29,8 +29,97 @@ def construct(self, *args):
 # Transforms
 
 
+class ImageTransform:
+    """
+    Tracks transformations that are performed on an image.
+
+    This allows multiple transforms to be processed in a single operation and also
+    provides a matrix that can be used to transform the focal point of the image.
+    """
+    def __init__(self, size):
+        self._check_size(size)
+        self.size = size
+        self.scale = (1.0, 1.0)
+        self.offset = (0.0, 0.0)
+
+    def clone(self):
+        clone = ImageTransform(self.size)
+        clone.scale = self.scale
+        clone.offset = self.offset
+        return clone
+
+    def resize(self, size):
+        """
+        Change the image size, stretching the transform to make it fit the new size.
+        """
+        self._check_size(size)
+        clone = self.clone()
+        clone.scale = (
+            clone.scale[0] * size[0] / self.size[0],
+            clone.scale[1] * size[1] / self.size[1]
+        )
+        clone.size = size
+        return clone
+
+    def crop(self, rect):
+        """
+        Crop the image to the specified rect.
+        """
+        self._check_size(tuple(rect.size))
+
+        # Transform the image so the top left of the rect is at (0, 0), then set the size
+        clone = self.clone()
+        clone.offset = (
+            clone.offset[0] - rect.left / self.scale[0],
+            clone.offset[1] - rect.top / self.scale[1]
+        )
+        clone.size = tuple(rect.size)
+        return clone
+
+    def transform_vector(self, vector):
+        """
+        Transforms a vector from the image's coordinate space.
+
+        Returns a new vector.
+        """
+        return Vector(
+            (vector.x + self.offset[0]) * self.scale[0],
+            (vector.y + self.offset[1]) * self.scale[1]
+        )
+
+    def untransform_vector(self, vector):
+        """
+        Transforms a back to the images coordinate space.
```

Comment:
A few words missing here. In addition, in this and the `transform_vector` docstring, I think it's worth noting what space it's being transformed to (presumably real space?)